From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Dohrmann <erbse.13@gmx.de>
Date: Fri, 11 Oct 2024 10:49:27 +0000
Subject: [PATCH] increase min granularity for memfd

If the granularity is too small, we run into kvm memslot limits when
mapping devices using VFIO. Increase the limit to reduce the number of
required memslots.
Ideally this should be configurable, but for now, we always use 2 MiB.
---
 system/guest-memfd-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/system/guest-memfd-manager.c b/system/guest-memfd-manager.c
index c728ebafafd1127d317e0467a459567ac0a97784..51ac6476763794675c95e90054e4da0734072fcf 100644
--- a/system/guest-memfd-manager.c
+++ b/system/guest-memfd-manager.c
@@ -149,7 +149,7 @@ static uint64_t guest_memfd_rdm_get_min_granularity(const RamDiscardManager *rdm
     GuestMemfdManager *gmm = GUEST_MEMFD_MANAGER(rdm);
 
     g_assert(mr == gmm->mr);
-    return gmm->block_size;
+    return 0x200000;
 }
 
 static void guest_memfd_rdm_register_listener(RamDiscardManager *rdm,
